### DISCUSSION TITLE
Repository language(s) detection not working

### DISCUSSION THREAD

**Discussion by jalbiero** [community/community#22788] [New to GitHub]
<p>My repository has octave/matlab and Makefiles files. None of them are recognized by Github (repo information say nothing about those languages and text editor/viewers don’t have color syntax).  What should I do in order to allow that? I worked in a lot of projects, I have never faced this problem with other programming languages.</p>

**Comment by airtower-luna** [comment]
<p>It’s hard to give a specific answer here, I have repositories where both are recognized correctly. What I can say is: Github uses Linguist to detect the languages of your files. The README contains a lot of details on how it detects languages and troubleshooting options. And if after that it looks like a bug you can open an issue there.<br>
</p><aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/github/linguist" target="_blank" rel="noopener">GitHub</a>
  </header>
  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/5814919/181073135-62502501-41e9-4ef1-8b44-783b2905d056.png" class="thumbnail onebox-avatar" width="400" height="400">

<h3><a href="https://github.com/github/linguist" target="_blank" rel="noopener">github/linguist</a></h3>

<p>Language Savant. If your repository's language is being reported incorrectly, send us a pull request! - github/linguist</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by jalbiero** [comment]
<p>Thanks for your quick answer <a class="mention" href="/airtower-luna">@airtower-luna</a>. I will read the README file before reporting a bug.  Thanks again :grinning:</p>

**Comment by jalbiero** [comment, marked as answer]
<p>The only way <em>github linguitst</em> detects a matlab/octave file (besides the extensions .matlab/.m) is if the file includes comments that starts with <strong>%</strong> (my files use the recommended <strong>#</strong> for octave). Check this file for more information:</p>
<aside class="onebox githubblob">
  <header class="source">
      <a href="https://github.com/github/linguist/blob/3d39c1c1a0d34c1af890ecfd18cf9ed595c68132/lib/linguist/heuristics.yml#L271" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>
  <article class="onebox-body">
    <h4><a href="https://github.com/github/linguist/blob/3d39c1c1a0d34c1af890ecfd18cf9ed595c68132/lib/linguist/heuristics.yml#L271" target="_blank" rel="noopener nofollow ugc">github/linguist/blob/3d39c1c1a0d34c1af890ecfd18cf9ed595c68132/lib/linguist/heuristics.yml#L271</a></h4>
<pre class="onebox"><code class="lang-yml"><ol class="start lines" start="261" style="counter-reset: li-counter 260 ;">
<li>    pattern: ':- module'</li>
<li>  - language: MUF</li>
<li>    pattern: '^: '</li>
<li>  - language: M</li>
<li>    pattern: '^\s*;'</li>
<li>  - language: Mathematica</li>
<li>    and:</li>
<li>      - pattern: '\(\*'</li>
<li>      - pattern: '\*\)$'</li>
<li>  - language: MATLAB</li>
<li class="selected">    pattern: '^\s*%'</li>
<li>  - language: Limbo</li>
<li>    pattern: '^\w+\s*:\s*module\s*{'</li>
<li>- extensions: ['.mask']</li>
<li>  rules:</li>
<li>  - language: Unity3d Asset</li>
<li>    pattern: 'tag:unity3d.com'</li>
<li>- extensions: ['.md']</li>
<li>  rules:</li>
<li>  - language: Markdown</li>
<li>    pattern:</li>
</ol></code></pre>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>I changed all the comments in my source files and now Github detects the repository language as MATLAB.</p>
<p>Maybe a pull request for the mentioned file in order to correct the regex will be the best solution.</p>

**Comment by milaniez** [comment]
Best way is for GitHub to determine what language an ambiguous file is by parsing the file content
